src/decode.rs

```diff
@@ -3838,14 +3838,11 @@ unsafe fn setup_tile(
         } else {
             ptr::null_mut()
         };
-        ts.frame_thread[p].cf = if !f.frame_thread.cf.is_null() {
-            f.frame_thread
-                .cf
-                .cast::<u8>()
-                .offset(
-                    (tile_start_off * size_mul[0] as usize >> (seq_hdr.hbd == 0) as c_int) as isize,
-                )
-                .cast::<DynCoef>()
+        ts.frame_thread[p].cf = if !f.frame_thread.cf.is_empty() {
+            f.frame_thread.cf
+                [(tile_start_off * size_mul[0] as usize >> (seq_hdr.hbd == 0) as c_int) as usize..]
+                .as_ptr()
+                .cast::<DynCoef>() as *mut _
```

Comment:
This doesn't work because `f` is a `&Rav1dFrameData`, so we can't get mutable access to `cf` in order to call `as_mut_ptr()`. I tried reworking the call site to pass a `&mut Rav1dFrameData` instead, but there are borrowing issues in the calling function (`rav1d_decode_frame_init_cdf`) that prevent that from working.

Reply:
Ah, okay.  We can fix them later then.